mojo/stdlib/stdlib/collections/string/_unicode.mojo

```diff
@@ -175,21 +176,41 @@ fn to_lowercase(s: StringSlice[mut=False]) -> String:
         A new string where cased letters have been converted to lowercase.
     """
     var data = s.as_bytes()
-    var result = String(capacity=_estimate_needed_size(len(data)))
+    # lowercased strings always have the same amount of bytes
+    var result = String(capacity=len(data))
     var input_offset = 0
     while input_offset < len(data):
+        alias `a` = Byte(ord("a"))
+        alias `A` = Byte(ord("A"))
+        alias lower_ascii_latin1 = `A` ^ `a`
+
         var rune_and_size = Codepoint.unsafe_decode_utf8_codepoint(
             data[input_offset:]
         )
-        var lowercase_char_opt = _get_lowercase_mapping(rune_and_size[0])
-        if lowercase_char_opt is None:
-            result.write_bytes(
-                data[input_offset : input_offset + rune_and_size[1]]
-            )
+        ref rune = rune_and_size[0]
+        ref size = rune_and_size[1]
+
+        if likely(size == 1):  # ASCII fast path
```

Comment:
Is there a perf difference when using `likely` rather than when just using the branch as is?
Not sure if it's worth adding here unless there is a measurable difference. 
In C++ the `[[unlikely]] / [[likely]]` attributes were added as well but using them can sometimes lead to weird perf implications so better to just have this verified here :)